In [1]:
from classes import Paths

import pandas as pd
import numpy as np

In [2]:
paths = Paths()
path = paths.driver_info
driver_info = pd.read_csv(path)
driver_info.sample(10)

age  user_rating  user_rides  user_time_accident     user_id  sex  \
13685   32          8.6        1014                20.0  P16847984W    1   
11004   44          8.5        1119                 NaN  J16557495M    0   
13202   47          7.5         642                 6.0  e19257942i    0   
12409   18          8.5         633                 6.0  M55314014B    1   
7031    43          8.6         424                 NaN  P15346566n    0   
347     42          8.9        1767                16.0  l17188711P    0   
6116    31          7.8         839                 2.0  q21309633s    1   
9255    40          8.9        1114                12.0  s14811318S    0   
5293    39          8.3         856                 6.0  Q31949393C    0   
1747    29          8.5        1147                 6.0  t11235313m    1   

      first_ride_date  
13685       2020-11-7  
11004       2019-1-23  
13202       2019-5-27  
12409       2019-5-29  
7031        2019-7-24  
347         2021-8-18  
6116        2020-9-20  
9255        2021-2-27  
5293         2019-4-4  
1747       2020-12-11

In [3]:
driver_info.describe()

age   user_rating    user_rides  user_time_accident  \
count  15153.000000  15153.000000  15153.000000        14182.000000   
mean      33.581073      8.109345    871.015442           10.129601   
std        9.919271      0.668246    573.512378            8.867482   
min       18.000000      5.800000      0.000000            0.000000   
25%       25.000000      7.700000    421.000000            4.000000   
50%       33.000000      8.100000    876.000000            9.000000   
75%       41.000000      8.600000   1145.000000           14.000000   
max       63.000000     10.000000   3316.000000           86.000000   

                sex  
count  15153.000000  
mean       0.499967  
std        0.500016  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        1.000000  
max        1.000000

<div class='alert alert-warning'>
1 What is the median age of drivers?
</div>

In [4]:
driver_info['age'].median()

33.0

<div class="alert alert-success">

**Answer:** Median age of drivers is 33

</div>

<div class='alert alert-warning'>
2 What is the minimum rating of the driver? (user_rating).
</div>

In [5]:
driver_info['user_rating'].min()

5.8

<div class="alert alert-success">

**Answer:** Minimum user (driver) rating is 5.8

</div>

<div class='alert alert-warning'>
3 What is the most frequent number of incidents among drivers? (user_time_accident).
</div>

In [6]:
driver_info['user_time_accident'].mode()

# or we can use groupby method or value_counts for detailed information
# driver_info['user_time_accident'].value_counts()
# driver_info.groupby("user_time_accident")["user_id"].count().nlargest(1)

0    6.0
Name: user_time_accident, dtype: float64

<div class="alert alert-success">

**Answer:** The most frequent number of incidents among drivers is 6

</div>

<div class='alert alert-warning'>
4 How many rows are there in the dataset?
</div>

In [7]:
print("Dataframe has:\nRows: {:10}\nColumns: {:7}".
        format(driver_info.shape[0], driver_info.shape[1]))

Dataframe has:
Rows:      15153
Columns:       7


<div class="alert alert-success">

**Answer:** Dataframe has 15153 rows

</div>

<div class='alert alert-warning'>
5 What is the most popular gender?
</div>

In [8]:
driver_info['sex'].value_counts()
# driver_info.sex.value_counts().nlargest(1)

sex
0    7577
1    7576
Name: count, dtype: int64

<div class="alert alert-success">

**Answer:** The most popular gender is 0, but to be more precise we have ~%50 of each gender in users base

</div>

